sim/src/analytics.rs

```diff
@@ -43,6 +43,16 @@ pub struct Analytics {
     // TODO Hack: No TripMode means cancelled
     // Finish time, ID, mode, trip duration
     pub finished_trips: Vec<(Time, TripID, Option<TripMode>, Duration)>,
+
+    // Records how long was spent waiting at each turn (Intersection) for a given trip
+    // Over a certain threshold
+    // TripID, [(TurnID, Time Waiting In Seconds)]
+    pub trip_intersection_delays: BTreeMap<TripID, Vec<(TurnID, u8)>>,
```

Comment:
Per the comment in `info/trip.rs` about searching through the `Vec` for every step in the path, I think `BTreeMap<(TripID, TurnID), u8>` might be more straightforward

Reply:
Do you think a tuple of keys would be better, or a nested BTreeMap?
Like: `BTreeMap<TripID, BTreeMap<TurnID, u8>>`

Reply:
It depends on the access patterns. In this case, we only want to look up by both trip and turn. If we also wanted to look up just by trip, the nested map would be useful.